app/models/miq_worker_type.rb

```diff
@@ -0,0 +1,54 @@
+class MiqWorkerType < ApplicationRecord
+  EXCLUDED_CLASS_NAMES = %w[ManageIQ::Providers::BaseManager::OperationsWorker].freeze
+
+  KILL_PRIORITY_METRICS_PROCESSOR_WORKERS = 10
+  KILL_PRIORITY_METRICS_COLLECTOR_WORKERS = 20
+  KILL_PRIORITY_REPORTING_WORKERS         = 30
+  KILL_PRIORITY_SMART_PROXY_WORKERS       = 40
+  KILL_PRIORITY_GENERIC_WORKERS           = 50
+  KILL_PRIORITY_EVENT_HANDLERS            = 60
+  KILL_PRIORITY_REFRESH_WORKERS           = 70
+  KILL_PRIORITY_SCHEDULE_WORKERS          = 80
+  KILL_PRIORITY_PRIORITY_WORKERS          = 90
+  KILL_PRIORITY_WEB_SERVICE_WORKERS       = 100
+  KILL_PRIORITY_REFRESH_CORE_WORKERS      = 110
+  KILL_PRIORITY_VIM_BROKER_WORKERS        = 120
+  KILL_PRIORITY_EVENT_CATCHERS            = 130
+  KILL_PRIORITY_UI_WORKERS                = 140
+  KILL_PRIORITY_REMOTE_CONSOLE_WORKERS    = 150
+  KILL_PRIORITY_COCKPIT_WS_WORKERS        = 160
+
+  scope :in_kill_order, -> { order(:kill_priority => :asc) }
+
+  def self.seed
+    transaction do
+      clean_worker_types
+      classes_for_seed.each { |klass| seed_worker(klass) }
+    end
+  end
+
+  def self.worker_class_names
+    pluck(:worker_type)
+  end
+
+  def self.worker_class_names_in_kill_order
+    in_kill_order.pluck(:worker_type)
+  end
+
+  private_class_method def self.classes_for_seed
+    @classes_for_seed ||= MiqWorker.descendants.select { |w| w.subclasses.empty? } - EXCLUDED_CLASS_NAMES.map(&:constantize)
```

Comment:
I forget, will this autoload all workers in subclasses in external repos such as provider workers?

I wonder what will happen here when/if we dissect bundler groups into small components and therefore won't all the worker classes.  But that's for another day.

Reply:
Yes, it will autoload all the provider subclasses.

If we split out the bundler groups for providers I would imagine the server would need all of the groups to be able to deal with provider constants, right? I don't think any of the workers with pared down bundler groups would be calling this method.